### ISSUE TITLE
Improve translate ?

### ISSUE THREAD

**Issue by bristow** [Fr4gorSoftware/SecScanQR#77]
Hi !

I'm french and i can translate the last strings. How to do that ?

Thanks

**Comment by Fr4gorSoftware** [comment]
Hi, thank you for the offer. Copy the xml code below in your favorite text editor and translate the strings. After that post the complete xml code as response (You need to use "insert code"). I will do the rest for you. 
example:
  before:
  `<string name="activity_generate">QR-Code Generator</string>`
  after:
  `<string name="activity_generate">Générer QR-Code</string>`

 ```
<resources>
    <!--Activity labels -->
    <string name="app_name">SecScanQR</string>
    <string name="activity_generate">QR-Code Generator</string>
    <string name="activity_history">History</string>
    <string name="activity_history_details">History</string>
    <string name="title_activity_geo_generator">Geo Generator</string>
    <string name="title_activity_text_generator">Text Generator</string>
    <string name="title_activity_barcode_generate">Barcode Generator</string>
    <string name="title_activity_vcard_generator">VCard Generator</string>
    <!-- MainActivity-->
    <string name="title_format">Code format:</string>
    <string name="title_information">Code information:</string>
    <string name="default_text_main_activity">Press on the "Scan" Button to scan a QR-Code/Barcode</string>
    <string name="send_to">Share with:</string>
    <string name="xzing_label">Activate the flashlight using the volume keys</string>
    <string name="notice_clipoard">Copied to Clipboard</string>
    <!--MainActivity Errors-->
    <string name="error_scan_first">Scan a QR-Code first</string>
    <string name="error_canceled_scan">You canceled the scanning</string>
    <string name="error_add_to_database">Was not able to add QR-Code to the history</string>
    <!-- MainActivity Navigation-->
    <string name="title_scan">Scan</string>
    <string name="title_generate">Generate</string>
    <string name="title_history">History</string>
    <!-- MainActivity Action Navigation-->
    <string name="title_copy">Copy</string>
    <string name="title_share">Share</string>
    <string name="title_reset">Reset</string>
    <string name="title_openInWeb">Open in Web</string>

    <!-- GenerateActivity-->
    <string name="btn_save">Save</string>
    <string name="toast_save">Saved in "SecScanQR" folder on the internal storage</string>
    <string name="toast_permission_needed">Permission is needed</string>
    <string name="toast_erorr">An error had occur</string>


    <!-- BarcodeGeneratorActivity-->
    <string name="activity_barcode_generator">Barcode Generator</string>


    <!-- TextGeneratorActivity-->
    <string name="hint_text2qr">Here you can enter some text</string>
    <string name="tv_format">Format: </string>

    <!-- TextGeneratorActivity Erorrs-->
    <string name="error_text_first">First type in some Text</string>
    <string name="error_generate">No valid input for this format</string>

    <!-- GeoGeneratorActivity-->
    <string name="hint_latitude">latitude example: 43.73</string>
    <string name="hint_longtitude">longtitude example: -10.73</string>

    <string name="cbLatitude">north</string>
    <string name="cbLongtitude">east</string>

    <!-- GeoGeneratorActivity Errors-->
    <string name="error_geo_first">First fill in valid values for the latitude and longtitude</string>

    <!-- VCARDGeneratorActivity-->
    <string name="hint_name">name</string>
    <string name="hint_fn">first name</string>
    <string name="hint_org">organization</string>
    <string name="hint_web">website</string>
    <string name="hint_email">e-mail</string>
    <string name="hint_mobil">mobil</string>
    <string name="hint_tele_work">telephone(work)</string>
    <string name="hint_tele_private">telephone(private)</string>
    <string name="hint_street">street</string>
    <string name="hint_plz">postcode</string>
    <string name="hint_city">city</string>
    <string name="hint_state">state</string>
    <string name="hint_country">country</string>

    <!-- VCARDGeneratorActivity Errors-->
    <string name="error_fn_or_name_first">First type in a name or first name</string>

    <!-- HistoryActivity -->
    <string name="reset_database">Reset</string>
    <string name="share_database">Share</string>
    <!-- HistoryActivity Delete Dialog -->
    <string name="delete_history_dialog_message">All entries will be deleted!</string>
    <string name="delete_history_dialog_confirmation">OK</string>
    <string name="delete_history_dialog_cancel">Cancel</string>

    <!-- HistoryActivity Errors-->
    <string name="error_not_in_database">Was not found in the history database</string>
    <string name="error_sqlexception">This format is not supported for the history feature</string>

    <!-- HistoryDetailsActivity-->
    <string name="notice_deleted_from_database">Deleted from the History</string>
    <!-- HistoryDetailsActivity Action Navigation-->
    <string name="title_delete">Delete</string>

    <!-- Optionsmenu-->
    <string name="settings_menu_about">About</string>
    <string name="settings_settings">Settings</string>

    <!-- AboutDialog-->
    <string name="about_dialog">About</string>
    <string name="copyright">Copyright Notice</string>
    <string name="copyright_holder">Copyright© 2017-2019 T-Dankworth</string>
    <string name="license">This App was created under the terms of the GNU General Public License. For more information please check out the link below: https://www.gnu.org/licenses/gpl-3.0.en.html</string>
    <string name="logo_designer">The App logo was created by s-yous: https://github.com/s-yous</string>
    <string name="translator">English translation by Thore Dankworth: https://github.com/Fr4gorSoftware</string>
    <string name="zxing_license">The App uses the ZXING Library which is under the Apache 2.0 license. For more information please check out the link below: http://www.apache.org/licenses/LICENSE-2.0</string>
    <!-- Strings related to Settings -->
    <string name="title_activity_settings">Settings</string>
    <!-- Scanner settings -->
    <string name="pref_header_scanner">Scanner</string>
    <string name="pref_header_history_settings">History</string>
    <string name="pref_header_search_engine_settings">Search Engine</string>

    <string name="pref_title_camera">Camera</string>
    <string-array name="pref_camera_entries">
        <item>back camera</item>
        <item>front camera</item>
    </string-array>
 

    <string name="pref_title_auto_scan">Autostart scanner</string>
    <string-array name="pref_auto_scan_entries">
        <item>On</item>
        <item>Off</item>
    </string-array>


    <string name="pref_title_auto_clipboard">Smart clipboard</string>
    <string-array name="pref_auto_clipboard_entries">
        <item>On</item>
        <item>Off</item>
    </string-array>


    <string name="pref_title_day_night_mode">Day/Night mode (ALPHA Version)</string>
    <string-array name="pref_day_night_mode_entries">
        <item>Day (Needs a restart)</item>
        <item>Night (Needs a restart)</item>
    </string-array>

    <string name="pref_title_history">History (on/off)</string>
    <string-array name="pref_history_entries">
        <item>On</item>
        <item>Off</item>
    </string-array>

    <string name="pref_title_search_engine">Search Engine</string>
    <string-array name="pref_search_engine_entries">
        <item>Bing</item>
        <item>DuckDuckGo</item>
        <item>Google</item>
        <item>Qwant</item>
        <item>Qwant Lite</item>
        <item>Startpage</item>
        <item>Yahoo</item>
        <item>Yandex (RUS)</item>
    </string-array>

    <string name="pref_title_barcode_search_enigne">Barcode Search Engine</string>
    <string-array name="pref_barcode_search_engine_entries">
        <item>No extra search engine for Barcodes</item>
        <item>Open Food Facts</item>
        <item>CodeCheck</item>
    </string-array>

</resources>
```

**Comment by bristow** [comment]
Here is the french strings

[strings_fr.zip](https://github.com/Fr4gorSoftware/SecScanQR/files/2901843/strings_fr.zip)

I took the XML file from here : https://github.com/Fr4gorSoftware/SecScanQR/blob/master/app/src/main/res/values/strings.xml

Hope it's ok !

**Comment by Fr4gorSoftware** [comment]
looks good. Will be implemented for version 1.2.1
